### ISSUE TITLE
Let `hover_docs` act like a toggle

### ISSUE THREAD

**Issue by kaste** [gladclef/HoverDocs#4]
That's an early feature request.  I think it would be really cool if one key binding would be enough -- t.i. we wouldn't need a different binding to close the popup/annotation.

For example, consider `  { "keys": ["ctrl+q"], "command": "hover_docs"},`,  now `ctrl+q` opens the popup, and a subsequent `ctrl+q` closes the popup *iff* we're still on the same token, or if the popup would be identical to the one already open (same position, same content).  Otherwise the popup/annotation gets closed and a new one appears (it gets replaced) just like now.

**Comment by gladclef** [comment]
Unfortunately I'm not aware of a way to do this reliably. Sublime doesn't have an API for determining what popups or annotations are currently visible. I would be toggling based on the last known state, and it's possible that that the popups would have been closed through some other means (eg by scrolling out of the view) and so the keyboard shortcut would try to "remove" a popup that no longer exists.

Basically, I think the better experience with these limitations is to leave it as is. IDK, what do you think?

**Comment by kaste** [comment]
The tooltips have an "on-close"[1] callback so you actually can record if a tooltip has been closed (for whatever reasons).  I did something along that in GitSavvy (although not very polished code)  https://github.com/timbrel/GitSavvy/blob/8ac1bdc2aad00e5630b03f9c52eb6908ab6e63d8/core/utils.py#L138-L164

EDIT: [1] it is actually `on_hide`

**Comment by gladclef** [comment]
I was not aware of that. Thanks for pointing it out!
